test/cljam/io/sam_test.clj

```diff
@@ -309,6 +310,43 @@
         "temp.baam"
         "temp.bai"))))
 
+(deftest writer-index-option-test
+  (testing "sam"
+    (with-before-after {:before (prepare-cache!)
+                        :after (clean-cache!)}
+      (with-open [w (sam/writer temp-sam-file false)]
+        (sam/write-header w (:header test-sam-sorted-by-pos))
+        (sam/write-refs w (:header test-sam-sorted-by-pos))
+        (sam/write-alignments w (:alignments test-sam-sorted-by-pos) (:header test-sam-sorted-by-pos)))
+      (is (not (.exists (cio/file (str temp-sam-file ".bai")))))))
+  (testing "bam"
+    (with-before-after {:before (prepare-cache!)
+                        :after (clean-cache!)}
+      (with-open [w (sam/writer temp-bam-file false)]
+        (sam/write-header w (:header test-sam-sorted-by-pos))
+        (sam/write-refs w (:header test-sam-sorted-by-pos))
+        (sam/write-alignments w (:alignments test-sam-sorted-by-pos) (:header test-sam-sorted-by-pos)))
+      (is (not (.exists (cio/file (str temp-bam-file ".bai")))))
+      (with-open [w (sam/writer temp-bam-file true)]
+        (sam/write-header w (:header test-sam))
+        (sam/write-refs w (:header test-sam))
+        (sam/write-alignments w (:alignments test-sam) (:header test-sam)))
+      (is (not (.exists (cio/file (str temp-bam-file ".bai")))))
+      (with-open [w (sam/writer temp-bam-file true)]
+        (sam/write-header w (:header test-sam-sorted-by-qname))
+        (sam/write-refs w (:header test-sam-sorted-by-qname))
+        (sam/write-alignments w (:alignments test-sam-sorted-by-qname) (:header test-sam-sorted-by-qname)))
+      (is (not (.exists (cio/file (str temp-bam-file ".bai")))))
+      (with-open [w (sam/writer temp-bam-file true)]
+        (sam/write-header w (:header test-sam-sorted-by-pos))
+        (sam/write-refs w (:header test-sam-sorted-by-pos))
+        (sam/write-alignments w (:alignments test-sam-sorted-by-pos) (:header test-sam-sorted-by-pos)))
+      (is (.exists (cio/file (str temp-bam-file ".bai"))))))
```

Comment:
Check if the content of the BAI is correct

Reply:
Test added.